zmon_worker_monitor/builtins/plugins/scalyr.py

```diff
@@ -95,29 +95,13 @@ def logs(self, query, max_count=100, minutes=5, continuation_token=None, columns
         else:
             raise CheckError('No logs or error message was returned from scalyr')
 
-    def function(self, function, query, minutes=5, end=0):
-
-        val = {
-            'token': self.__read_key,
-            'queryType': 'numeric',
-            'filter': query,
-            'function': function,
-            'startTime': parse_timestamp(minutes),
-            'priority': 'low',
-            'buckets': 1
-        }
-        if end is not None:
-            val['endTime'] = parse_timestamp(end)
-
-        r = requests.post(self.__numeric_url, json=val, headers={'Content-Type': 'application/json'})
-
-        r.raise_for_status()
-
-        j = r.json()
-        if 'values' in j:
-            return j['values'][0]
-        else:
-            return j
+    def function(self, function, query, minutes=5, end=0, align=0):
+        value = self.timeseries(filter=query, function=function, minutes=minutes, end=end, align=align)
+        if isinstance(value, int):
+            return value
+        if isinstance(value, list):
```

Comment:
When would this return a list?

Reply:
I just added the list case because:
* [The existing tests had list-data](https://github.com/zalando-zmon/zmon-worker/blob/master/tests/test_scalyr.py#L184)
* [Technically the timeseries function can return a list](https://github.com/zalando-zmon/zmon-worker/blob/master/zmon_worker_monitor/builtins/plugins/scalyr.py#L178)

We can remove the list case and change the test-data (thumbs up if you'd prefer that)